dbaccess/transaction.go

```diff
@@ -182,6 +182,39 @@ func AcceptedTransactionsByBlockID(ctx database.Context, blockID uint64, preload
 	return transactions, nil
 }
 
+// TransactionDoublespends retrieves transactions, that have at least one input which is the same as in transaction that has the provided hash
+func TransactionDoublespends(ctx database.Context, transactionHash string, preloadedFields ...dbmodels.FieldName) ([]*dbmodels.Transaction, error) {
+	db, err := ctx.DB()
+	if err != nil {
+		return nil, err
+	}
+
+	origInputs := db.Model(&dbmodels.TransactionInput{}).
+		Column("transaction_input.transaction_id", "transaction_input.index").
+		Join("INNER JOIN transactions").
+		JoinOn("transactions.id = transaction_input.transaction_id").
+		Where("transactions.transaction_hash = ?", transactionHash)
```

Comment:
transaction_id instead of transaction_hash.

Note: There might be some confusion between the transaction_id column in other tables (which refer's to the transaction's in-database id) and the transaction's id (which is the ID by which other transactions in the DAG refer to it).
Note that unlike bitcoin, in kaspa the transaction hash and transaction ID are different things.